sky/jobs/server/core.py

```diff
@@ -366,6 +396,18 @@ def queue(refresh: bool, skip_finished: bool = False) -> List[Dict[str, Any]]:
                            f'{returncode}')
 
     jobs = managed_job_utils.load_managed_job_queue(job_table_payload)
+
+    if not all_users:
+        # For backwards compatibility, we show jobs that have user_hash None.
+        def user_hash_matches_or_missing(job: Dict[str, Any]) -> bool:
+            if 'user_hash' not in job:
+                return True
+            if job['user_hash'] is None:
+                return True
+            return job['user_hash'] == common_utils.get_user_hash()
+
+        jobs = list(filter(user_hash_matches_or_missing, jobs))
```

Comment:
minor: it is more performant to directly apply the filtering in the SQL oepration. Not urgent

Reply:
Did it this way for simplicity and to match what we do for `sky status`.